src/sagas/templatesSaga.js

```diff
@@ -2,29 +2,34 @@ import { TemplateLibrary } from '@accordproject/cicero-core';
 import { version as ciceroVersion } from '@accordproject/cicero-core/package.json';
 
 import { takeLatest, put } from 'redux-saga/effects';
+import * as actions from '../actions/templatesActions';
 
+/**
+ * saga to populate store with templates
+ */
 export function* pushTemplatesToStore() {
   try {
     const templateLibrary = new TemplateLibrary();
     const templateIndex = yield templateLibrary
       .getTemplateIndex({ latestVersion: false, ciceroVersion });
     const templateIndexArray = Object.values(templateIndex);
-    yield put({ type: 'AP_TEMPLATES_RECEIVED', templates: templateIndexArray });
+    yield put(actions.getTemplatesSuccess(templateIndexArray));
   } catch (err) {
-    yield put({ type: 'AP_TEMPLATES_ERROR', error: err });
+    yield put(actions.getTemplatesError(err));
   }
 }
 
+/**
+ * saga to add new template to store
```

Comment:
The comments could be more of a train of thought for the time being. `saga which puts a mock template onto the array of templates in the store`